uast/utils.go

```diff
@@ -66,13 +66,64 @@ func CyclomaticComplexity(n *Node)  int {
 		}
 		n := p.Node()
 		for _, r := range n.Roles {
-			switch(r) {
+			switch r {
 			case If, SwitchCase, For, ForEach, While,
-			     DoWhile, TryCatch, Continue, OpBooleanAnd, OpBooleanOr,
-			     OpBooleanXor:
+				DoWhile, TryCatch, Continue, OpBooleanAnd, OpBooleanOr,
+				OpBooleanXor:
 				complx++
 			}
 		}
 	}
 	return complx
 }
+
+func (n *Node) ContainsRole(role Role) bool {
```

Comment:
It would seem more idiomatic to change `Roles []Role` field to a `Roles` type:
```
type Roles []Role
```

and implement `Contains` in `Roles`. So code using it would do: `n.Roles.Contains(role)`.

At least if proteus behaves correctly with it when generating protobuf.